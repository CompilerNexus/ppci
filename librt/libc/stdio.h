#ifndef STDIO_H
#define STDIO_H

void printf(char*);
void bsp_putc(char);
void myprint(char* label, int num);

#endif

