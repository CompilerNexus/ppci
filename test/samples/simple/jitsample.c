
#include <stdio.h>

int x(int*a, int*b, int count)
{
  int sum;
  int i;
  sum = 0;
  for (i=0; i<count; i++)
    sum += a[i] * b[i];
  return sum;
}

void main_main()
{
  int a[3] = {1,2,3};
  struct {int foo; int b[3];} b = {1927, {5,4,9}};
  myprint("a=", a[2]);
  int count = 3;
  count = 3;
  int res;
  res = x(a, b.b, count);
  myprint("x=", res);
}
