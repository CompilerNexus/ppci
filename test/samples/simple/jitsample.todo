
#include <stdio.h>

int x(int*a, int*b, int count)
{
  int sum;
  int i;
  sum = 0;
  for (i=0; i<count; i++)
    sum += a[i] * b[i];
  return sum;
}

void main_main()
{
  int a[3] = {1,2,3}; // TODO
  a[0] = 1;
  a[1] = 2;
  a[2] = 3;
  int b[3] = {5,4,9}; // TODO!
  b[0] = 5;
  b[1] = 4;
  b[2] = 9;
  myprint("a=", a[2]);
  int count = 3;
  count = 3;
  int res;
  res = x(a, b, count);
  myprint("x=", res);
}
