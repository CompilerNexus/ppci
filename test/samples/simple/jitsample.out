a=00000003
x=00000028
